### PR TITLE
Switch to dtolnay rust-toolchain

### LINKED ISSUE #317: [TASK] Remove actions-rs actions from CI
## Motivation
Since action-rs is no longer maintained it will be better to opt for alternative solutions.
 For more details refer [action-rs/#2216](https://github.com/actions-rs/toolchain/issues/216)

## Solution
- Use [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain) instead of actions-rs/toolchain
- Use [rustsec/audit-check](https://github.com/rustsec/audit-check) (fork of actions-rs/audit-check) instead of actions-rs/audit-check
- Use `cargo` directly instead of actions-rs/cargo

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- Use [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain) instead of actions-rs/toolchain
- Use [rustsec/audit-check](https://github.com/rustsec/audit-check) (fork of actions-rs/audit-check) instead of actions-rs/audit-check
- Use `cargo` directly instead of actions-rs/cargo

### Reference issue to close (if applicable)
- Closes #317 

-----
### Code Checklist 

- [ ] Tested
- [ ] Documented

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -26,23 +26,18 @@ jobs:
...
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2
-
-      - uses: actions-rs/cargo@v1
-        with:
-          use-cross: true
-          args: --release --target=${{ matrix.target }} --features native-tls/vendored --locked
-          command: build
+      - name: Build target
+        run: cargo build --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```
Comment (dutterbutter) [inline .github/workflows/check.yml:40]: The intent would be to run `build` using `cross` 

```suggestion
      - name: Build target using cross
        run: cross build --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -108,14 +98,16 @@ jobs:
...
+
       - name: Install cross
         run: cargo install cross --locked
       - name: Build target
-        uses: actions-rs/cargo@v1
-        with:
-          use-cross: true
-          args: --release --target=${{ matrix.target }} --features native-tls/vendored --locked
-          command: build
+        run: cargo build --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```
Comment (dutterbutter) [inline .github/workflows/check.yml:110]: The intent would be to run `build` using `cross` 

```suggestion
      - name: Build target using cross
        run: cross build --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -64,23 +59,18 @@ jobs:
...
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2
-
-      - uses: actions-rs/cargo@v1
-        with:
-          use-cross: true
-          args: --release --target=${{ matrix.target }} --locked
-          command: test
+      - name: Build target
+        run: cargo test --release --target=${{ matrix.target }} --locked
```
Comment (dutterbutter) [inline .github/workflows/check.yml:73]:
```suggestion
      - name: Build target using cross
        run: cross test --release --target=${{ matrix.target }} --locked
```

File: .github/workflows/check.yml [hunk 4]
```diff
@@ -162,11 +160,7 @@ jobs:
       - name: Install cross
         run: cargo install cross --locked
       - name: Build target
-        uses: actions-rs/cargo@v1
-        with:
-          use-cross: true
-          args: --release --target=${{ matrix.target }} --features native-tls/vendored --locked
-          command: test
+        run: cargo test --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```
Comment (dutterbutter) [inline .github/workflows/check.yml:163]:
```suggestion
        run: cross test --release --target=${{ matrix.target }} --features native-tls/vendored --locked
```

**Review by shekohex — APPROVED** [review]
LGTM!

File: .github/workflows/audit.yml [hunk 5]
```diff
@@ -19,6 +19,6 @@ jobs:
           access_token: ${{ github.token }}
       - uses: actions/checkout@v2
 
-      - uses: actions-rs/audit-check@v1
+      - uses: rustsec/audit-check@master
```
Comment (dutterbutter) [inline .github/workflows/audit.yml:22]: Can  you link me this action @salman01zp ? I am not familiar with it.

**Review by dutterbutter — APPROVED** [review]
Reminder to update other Rust projects to remove deprecated `action-rs`

File: .github/workflows/audit.yml (hunk 5 above)
Comment (salman01zp) [reply .github/workflows/audit.yml:22]: https://github.com/rustsec/rustsec

**Pull request merged**